apisix/core/request.lua

```diff
@@ -131,26 +129,41 @@ function _M.set_header(ctx, header_name, header_value)
         changed = a6_request.is_request_header_set()
     end
 
-    ngx.req.set_header(header_name, header_value)
+    update_func(header_name, header_value)
```

Comment:
We can use override instead of passing an extra function.

Reply:
Fixed.